core/arch/atomic_exports.h

```diff
@@ -936,11 +936,12 @@ atomic_compare_exchange_int(volatile int *var, int compare, int exchange)
      * sequentially consistent memory order semantics.
      */
     __asm__ __volatile__("fence   iorw,ow\n\t"
-                         "1: lr.w.aq t0, (%1)\n\t"
-                         "   bne     t0, %2, 2f\n\t"
-                         "   sc.w.aq %0, %3, (%1)\n\t"
-                         "   bnez    %0, 1b\n\t"
-                         "2:"
+                         "1: lr.w.aqrl  t0, (%1)\n\t"
```

Comment:
Maybe add a comment on these suffixes  to lr and sc. What do they mean and why are they needed?

Reply:
It's basically what's described in the existing comment. But like what is said in spec[1], an `sc.w.aq` makes no sense. I changed the comment slightly in https://github.com/DynamoRIO/dynamorio/pull/6089/commits/3be7fd4d7b709206f05d900faa031773afc68cc5 to reflect this modification.

> Software should not set the rl bit on an LR instruction unless the aq bit is also set, nor should software set the aq bit on an SC instruction unless the rl bit is also set. LR.rl and SC.aq instructions are not guaranteed to provide any stronger ordering than those with both bits clear, but may result in lower performance.